docs/content/guides/limiting-concurrency-in-data-pipelines.mdx

```diff
@@ -0,0 +1,471 @@
+
+</TabItem>
+<TabItem name="Limiting specific runs using tags">
+
+### Limiting specific runs using tags
+
+How you limit specific runs based on tags depends on whether you're using Dagster Cloud or Dagster Open Source:
+
+- **Dagster Cloud**: Use the [Dagster Cloud UI or the dagster-cloud CLI][cloud-deployment-settings]
+- **Dagster Open Source**: Use your instance's `dagster.yaml`
+
+To enable this limit, use `run_queue.tag_concurrency_limits`. This key accepts a list of tags and their corresponding concurrency limits.
+
+```yaml file=/deploying/concurrency_limits/dagster.yaml startafter=start_tag_example endbefore=end_tag_example
+run_queue:
+  max_concurrent_runs: 15
+  tag_concurrency_limits:
+    - key: "database"
+      value: "redshift" # applies when the `database` tag has a value of `redshift`
+      limit: 4
+    - key: "dagster/backfill" # applies when the `dagster/backfill` tag is present, regardless of value
+      limit: 10
+```
+
+Let’s review what this configuration will do:
+
+- For runs with a `database` tag with a value of `redshift`, a maximum of four runs can execute concurrently
+- For runs with a `dagster/backfill` tag, a maximum of 10 runs can execute concurrently. Note that this implementation applies to **any** value of the `dagster/backfill` tag, whereas the `database: redshift` example only applies the limit when there’s a specific tag value.
+
+#### Limiting runs by unique tag value
+
+To apply separate limits to each unique value of a tag, set a limit for each unique value using `applyLimitPerUniqueValue`:
+
+```yaml file=/deploying/concurrency_limits/per-unique-value-dagster.yaml
+run_queue:
+  - key: "use-case"
+    value:
+      applyLimitPerUniqueValue: true
+    limit: 3
+```
+
+With this configuration, each unique value of the `use-case` tag will be limited to three concurrent runs. For example, `team: marketing` and `team: sales` will both be limited to three concurrent runs each.
+
+</TabItem>
+</TabGroup>
+
+---
+
+## Configuring op/asset-level concurrency
+
+<Note>
+  <strong>Looking for global op/asset concurrency?</strong> Dagster doesn’t
+  currently support global (across multiple runs) op/asset concurrency
+  out-of-the-box. If you require concurrency that applies to all ops/assets in a
+  deployment, check out our{" "}
+  <a href="/deployment/guides/celery">integration with Celery</a>.
+</Note>
+
+Utilizing op/asset-level concurrency provides fine-grained control for the maximum number of ops/assets that can be executed at once within a single run, ensuring that shared resources aren't overwhelmed.
+
+Op and asset concurrency can be configured by:
+
+- Limiting the number of ops/assets that can execute in a single run at once
+- Specifying limits using tags
+
+### Limiting overall concurrency in a job
+
+To limit concurrency for ops and assets in jobs, use `max_concurrent` in the job’s config, either in Python or using the Launchpad in the Dagster UI.
```

Comment:
this is where I think we should reference the executor (that's what's being configured in the "execution" key) - and will help explain why the key differs depending on what executor you're using

Reply:
I dunno if it's worth noting in an example that you can also apply this configuration directly on the executor itself via configured? That would let you supply a default across multiple jobs. Maybe we are de-emphasizing configured though since it's complicated? but it would look like this:

```
my_custom_executor = multiprocess_executor.configured({"max_concurrent": 2})
@job(executor=my_custom_executor)
...
```

Or even as a default in the Definitions object (which we don't currently show in the docs anywhere that you can do i don't think)

```
Definitions(
  ...
  default_executor_def=my_custom_executor
)
```

Reply:
I think this is good info, but I want to get these out sooner rather than later. I'm going to do a follow-up PR that adds this in.